tests/unit/services/cloudformation/test_change_set_describe_details.py

```diff
@@ -1222,3 +1245,192 @@ def test_condition_update_production_remove_resource(self):
             }
         ]
         self.compare_changes(changes, target)
+
+    def test_output_new_resource_and_output(self):
+        t1 = {
+            "Resources": {
+                "UnrelatedParam": {
+                    "Type": "AWS::SSM::Parameter",
+                    "Properties": {"Name": "unrelated-param", "Type": "String", "Value": "foo"},
+                }
+            }
+        }
+        t2 = {
+            "Resources": {
+                "UnrelatedParam": {
+                    "Type": "AWS::SSM::Parameter",
+                    "Properties": {"Name": "unrelated-param", "Type": "String", "Value": "foo"},
+                },
+                "NewParam": {
+                    "Type": "AWS::SSM::Parameter",
+                    "Properties": {"Name": "param-name", "Type": "String", "Value": "value-1"},
+                },
+            },
+            "Outputs": {"NewParamName": {"Value": {"Ref": "NewParam"}}},
+        }
+        outputs_unit = self.debug_outputs(t1, t2)
+        assert not outputs_unit.before_context
+        assert outputs_unit.after_context == [{"Name": "NewParamName", "Value": "NewParam"}]
```

Comment:
This is not correct. The value should be the `Name` of the parameter, which is autogenerated by CloudFormation. See the [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ssm-parameter.html#aws-resource-ssm-parameter-return-values). So it should be `param-name`.